### ISSUE TITLE
is it possible to view the password on saved connections? Thanks

### ISSUE THREAD

**Issue by andrub72** [mRemoteNG/mRemoteNG#2636]
is it possible to view the password on saved connections?
Thanks

**Comment by Kvarkas** [comment]
yes and now :) technically you could add external tool with a script who show you a password, but that will stop working once we implement a security patch. I am still thinking if allow users to decrypt passwords (many admins asking to dont!) but perhaps what will be possible under admin options only (in next release it will be normal user and elevated one, by default normal user will have all same permissions (for home users) but corporate admins will be able to restrict such.

**Comment by andrub72** [comment]
can you tell me currently what external tool or script I can use?
Thanks

Il giorno mar 27 ago 2024 alle ore 12:47 Dimitrij ***@***.***>
ha scritto:

**Comment by simonai1254** [comment]
This issue is a duplicate from #2598 


> Extracting the information from the connection file is generally considered leaking data, especially since mRemoteNG does not yet have proper encryption on the connection file. See https://github.com/mRemoteNG/mRemoteNG/issues/2420, and https://github.com/mRemoteNG/mRemoteNG/issues/2189.

> Anyway, if you really would like this feature you can help yourself with the guide of this comment here: https://github.com/mRemoteNG/mRemoteNG/issues/2193#issuecomment-1146343983

**Comment by andrub72** [comment]
Perfect ....
thank's

Il giorno mar 27 ago 2024 alle ore 13:30 Simon Monai <
***@***.***> ha scritto:

**Comment by Kvarkas** [comment]
> can you tell me currently what external tool or script I can use?

Step1: Create a new external tool by selecting ‘Tools’ -> ‘External Tools’, Right click and select ‘New External Tool’

![image](https://github.com/user-attachments/assets/d2309fc4-c337-4953-9891-a17a59175c4f)

Step2: Select New tab and fill with below details

![image](https://github.com/user-attachments/assets/e625e4f2-5f7f-4b21-bd03-3c99b6353e22)

Step3: Test the password by right clicking on you connection -> ‘External Tools’ -> ‘View Password’, the password details will open on cmd prompt screen.

![image](https://github.com/user-attachments/assets/f4bd1927-35c8-4059-b969-e44d07376ab8)

**Comment by andrub72** [comment]
Yes this .....

Create a new external tool by selecting ‘Tools’ -> ‘External Tools’, Right
click and select ‘New External Tool’

Il giorno mar 27 ago 2024 alle ore 16:13 Dimitrij ***@***.***>
ha scritto:

**Comment by Cool34000** [comment]
Prefer using as argument : `/k echo %!password%`
It will manage complex passwords (that have specials chars!)

My 2 cents

**Comment by Cool34000** [comment]
You can also use PowerShell to view passwords (better complex passwords management!)
Create your external tool that launches the script:
```
Name : View infos
File : powershell.exe
Args : -ExecutionPolicy Bypass -File "C:\Program Files (x86)\mRemoteNG\ShowInfos.ps1" -Description "%!description%" -Name "%!name%" -User "%!username%" -Pass "%!password%"
```

And the PowerShell script :
```
###################################################################################################################
# Paramsètres
param (
    [string]$Description = $null,
    [string]$Name        = $null,
    [string]$User        = $null,
    [string]$Pass        = $null
)
###################################################################################################################
# Fonctions
Function Pause {
    param (
        [string]$Color   = $null,
        [string]$Message = "Appuyez sur une touche pour continuer..."
    )
    if ($Color) {
        Write-Host "$Message" -ForegroundColor $Color
    } else {
        Write-Host "$Message"
    }
    $x = $host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")
}
###################################################################################################################
# Affichage des informations
$host.ui.RawUI.WindowTitle = "Voir les infos"
Write-Host
Write-Host "+-------------------------------------------------------"
Write-Host "| Client       : $Description"
Write-Host "| Equipement   : $Name"
Write-Host "| Utilisateur  : $User"
Write-Host "| Mot de passe : " -NoNewLine
Write-Host "$Pass" -ForegroundColor Yellow -BackgroundColor DarkRed
Write-Host "+-------------------------------------------------------"
Write-Host
Write-Host
if ($Pass -ne "") {
    Add-Type -AssemblyName System.Windows.Forms
    [System.Windows.Forms.Clipboard]::SetText($Pass)
    Write-Host "Info : " -NoNewLine -ForegroundColor Blue
    Write-Host "Le mot de passe a été copié dans votre presse papier !"
    Write-Host
}
Pause
###################################################################################################################
```

It will show all the infos and even copy the password for you in your clipboard!

**Comment by robertpopa22** [comment]
Triage refresh (P4 version-label debt): this issue is tagged with legacy version labels. Please retest on the latest build (fork v1.79.x or upstream dev), then share reproducible steps and expected vs actual behavior. If still valid, keep open and relabel to current triage scope (for example Need 2 check). If fixed, close with resolution.

Current legacy version labels: 1.77.3
